include/nuttx/spinlock.h

```diff
@@ -32,7 +32,7 @@
 
 #include <nuttx/irq.h>
 
-#ifndef CONFIG_SPINLOCK
```

Comment:
What was the reason of replacing this?

Reply:
The reason why use CONFIG_ARCH_HAVE_TESTSET instead of `CONFIG SPINLOCK` is : 
1 Our archs is xtensa, but not support sync instruction( S32C1L) in xtensa_testset.c.